Exercise_1.java

```diff
@@ -1,38 +1,80 @@
-class Stack { 
+
+public class Stack { 
     static final int MAX = 1000; 
     int top; 
     int a[] = new int[MAX]; // Maximum size of Stack 
   
-    boolean isEmpty() 
+    boolean isEmpty()   
     { 
         //Write your code here 
+    	if(top < 0)
+    	{
+    		System.out.println("Stack Underflow");
+    		return true;
+    	}
+		return false;
     } 
 
     Stack() 
     { 
-        //Initialize your constructor 
+        //Initialize your constructor
+    	top = -1;
     } 
   
     boolean push(int x) 
     { 
         //Check for stack Overflow
         //Write your code here
+    	if(top> MAX -1)
+    	{
+    		System.out.println("Stack is overflown");
+    		return false;
+    	}
+    	else
+    	{
+    		a[++top] =x;
+    		return true;
+    	}
+    		
+    	
     } 
   
     int pop() 
     { 
         //If empty return 0 and print " Stack Underflow"
+    	int popX =0;
         //Write your code here
+    	if(!isEmpty())
+    	{
+    		 popX = a[top--];
+    		 return popX;
+    	}
+    	else
+    	{
+    		isEmpty();

```

Comment:
There's no need to call isEmpty() in the else condition again

Reply:
Alright, got it. I will update it in my code.

Thank you!